### PR TITLE
add: implement event_handling_duration_seconds metric

### PR DESCRIPTION
**Pull request by sanposhiho**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

add: implement `event_handling_duration_seconds` metric.

#### Which issue(s) this PR fixes:

Part of #124566

#### Does this PR introduce a user-facing change?

```release-note
implement `event_handling_duration_seconds` metric, which is the time the scheduler takes to handle each kind of events. 
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by sanposhiho** [comment]
@gabesaba @macsko @alculquicondor can any of you possibly take a look.

File: pkg/scheduler/eventhandlers.go [hunk 1]
```diff
@@ -93,13 +99,25 @@ func (sched *Scheduler) updateNodeInCache(oldObj, newObj interface{}) {
...
+	updatingDuration := metrics.SinceInSeconds(start)
+
 	// Only requeue unschedulable pods if the node became more schedulable.
-	for _, evt := range nodeSchedulingPropertiesChange(newNode, oldNode) {
+	for _, evt := range events {
+		startMoving := time.Now()
 		sched.SchedulingQueue.MoveAllToActiveOrBackoffQueue(logger, evt, oldNode, newNode, preCheckForNode(nodeInfo))
+		movingDuration := metrics.SinceInSeconds(startMoving)
+
+		metrics.EventHandlingLatency.WithLabelValues().Observe(updatingDuration + movingDuration)
```
Comment (macsko) [inline pkg/scheduler/eventhandlers.go:113]: Shouldn't it have a label set?

**Review by macsko — COMMENTED** [review]
lgtm, one comment

And shouldn't the storage class update be measured [here](https://github.com/kubernetes/kubernetes/blob/25a3e81cee34d458b803e7b9475fe41c627ef21c/pkg/scheduler/eventhandlers.go#L552)?

**Comment by sanposhiho** [comment]
@macsko Both were good catches, updated.

**Comment by sanposhiho** [comment]
@kubernetes/sig-scheduling-approvers 
Please take another look for `/approve`.

File: pkg/scheduler/metrics/metrics.go [hunk 2]
```diff
@@ -83,6 +83,16 @@ var (
 			StabilityLevel: metrics.STABLE,
 		}, []string{"result", "profile"})
 
+	EventHandlingLatency = metrics.NewHistogramVec(
```
Comment (alculquicondor) [inline pkg/scheduler/metrics/metrics.go:86]: you forgot to add it to the list of registered metrics :)

File: pkg/scheduler/eventhandlers.go [hunk 3]
```diff
@@ -92,13 +98,25 @@ func (sched *Scheduler) updateNodeInCache(oldObj, newObj interface{}) {
...
+	updatingDuration := metrics.SinceInSeconds(start)
+
 	// Only requeue unschedulable pods if the node became more schedulable.
-	for _, evt := range queue.NodeSchedulingPropertiesChange(newNode, oldNode) {
+	for _, evt := range events {
+		startMoving := time.Now()
 		sched.SchedulingQueue.MoveAllToActiveOrBackoffQueue(logger, evt, oldNode, newNode, preCheckForNode(nodeInfo))
+		movingDuration := metrics.SinceInSeconds(startMoving)
+
+		metrics.EventHandlingLatency.WithLabelValues().Observe(updatingDuration + movingDuration)
```
Comment (alculquicondor) [inline pkg/scheduler/eventhandlers.go:112]: where is the label?

**Comment by sanposhiho** [comment]
@alculquicondor @macsko Fixed the points from Aldo (sorry looks like I made a mistake when solving a conflict)

**Pull request merged**